internal/peermanager/peerwantmanager.go

```diff
@@ -207,6 +238,9 @@ func (pwm *peerWantManager) sendCancels(cancelKs []cid.Cid) {
 		}
```

Comment:
We should be able to simplify all of this to:

(and remove the sets)

```go
if pwm.broadcastWants.Has(c) {
    broadcastCancels = append(broadcastCancels, c)
    pwm.wantGague.Dec()
} else if _, ok := pwm.wantPeers[c]; ok {
    pwm.wantGague.Dec()
}
```

Reply:
The problem is that we need to decrement the want-block gauge as well as the all-wants gauge, and we don't know if `pwm.wantPeers[c]` points to a want-have or a want-block.

Reply:
Ah, I see. Eh, then we can profile later to see if it makes a difference.